rosbag2_transport/test/rosbag2_transport/test_record.cpp

```diff
@@ -64,3 +64,20 @@ TEST_F(RecordIntegrationTestFixture, published_messages_from_multiple_topics_are
   EXPECT_THAT(array_messages[0]->bool_values, Eq(array_message->bool_values));
   EXPECT_THAT(array_messages[0]->float32_values, Eq(array_message->float32_values));
 }
+
+TEST_F(RecordIntegrationTestFixture, qos_is_stored_in_metadata)
+{
+  auto string_message = get_messages_strings()[1];
+  std::string topic = "/chatter";
+  start_recording({false, false, {topic}, "rmw_format", 100ms});
+  pub_man_.add_publisher<test_msgs::msg::Strings>(topic, string_message, 2);
+  run_publishers();
+  stop_recording();
+
+  MockSequentialWriter & writer =
+    static_cast<MockSequentialWriter &>(writer_->get_implementation_handle());
+  auto recorded_topics = writer.get_topics();
+  std::string serialized_profiles = recorded_topics.at(topic).offered_qos_profiles;
+  // Basic smoke test that the profile was serialized into the metadata as a string.
+  EXPECT_THAT(serialized_profiles, ContainsRegex("reliability"));
```

Comment:
Why not make this a bit more strict like the [`rosbag2 info` tests](https://github.com/ros2/rosbag2/blob/master/rosbag2_tests/test/rosbag2_tests/test_rosbag2_info_end_to_end.cpp#L43)?

Reply:
Yeah, good idea, I'll make it a little more strict